src/main/java/edu/harvard/iq/dataverse/api/DatasetFieldServiceApi.java

```diff
@@ -543,4 +536,86 @@ public static String getDataverseLangDirectory() {
     }
 
+    @POST
+    @Consumes("text/tab-separated-values")
+    @Path("mergeLanguageList")
+    public Response mergeLanguagePropertyFile(File file) {
+        ActionLogRecord alr = new ActionLogRecord(ActionLogRecord.ActionType.Admin, "mergeLanguageList");
+        alr.setInfo( file.getName() );
+        JsonArrayBuilder responseArr = Json.createArrayBuilder();
+
+        BufferedReader br = null;
+        String line = null;
+        String splitBy = "\t";
+        int lineNumber = 0;
+        int offset = 200; // number of existing languages // TODO: get the number from db
+
+        DatasetFieldType dsv = datasetFieldService.findByName("language");
+
+        try {
+            // Load the new language tab file
+            // The format of the file is: Id<tab>Part2b<tab>Part2t<tab>Part1<tab>Scope<tab>Language_Type<tab>Ref_Name<tab>Comment
+            // Example: aaa\t\t\t\tI\tL\tGhotuo
+            logger.fine("Loading ISOTabFile from " + file);
+
+            br = new BufferedReader(new FileReader("/" + file));
+            br.readLine(); // ignore first line as it is a header
+            while ((line = br.readLine()) != null) {
+                String[] fields = line.split(splitBy);
+                if (fields.length >= 7) {
+                    String code = fields[0];
+                    Set<String> codes = new HashSet<>();
+                    codes.addAll(List.of(code, fields[1], fields[2], fields[3]));
+                    codes.removeAll(Arrays.asList("", null));
+                    String name = fields[6];
+                    // get controlledvocabularyvalue by name or by codes
+                    ControlledVocabularyValue cvv = datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(dsv, name, true);
+                    // if not found, find using main code and alternate codes
+                    Iterator<String> codesIterator = codes.iterator();
+                    while (cvv == null && codesIterator.hasNext()) {
+                        cvv = datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(dsv, codesIterator.next(), true);
+                    }
+
+                    // if it is found we need to merge the alternate codes since the next step will delete all existing alternate codes before adding the new ones
+                    if (cvv != null) {
+                        cvv.getControlledVocabAlternates().stream().forEach(alt -> {
+                            codes.add(alt.getStrValue());
+                        });
+                    }
+
+                    // Now call parseControlledVocabulary to create/update the Controlled Vocabulary Language
+                    // values: unused, type, displayName, identifier, display order, alt codes...
+                    String displayName = cvv != null ? cvv.getStrValue() : name;
+                    String displayOrder = String.valueOf(cvv != null ? cvv.getDisplayOrder(): lineNumber + offset);
```

Comment:
It's hard to set the display order to remain alphabetical. Especially if the Dataverse installation isn't using English. There are 180 ish languages in the original tsv file. A sql script could be created to re-order them forcing the "Not applicable" last. The ISO639-3 file actually has a "zxx No linguistic content" entry at the end. It really isn't that noticable in the UI since it's filtered on the partial string you enter. I did reorder the tab file from the LoC since that too was ordered by the code and not the display string. I thought it was best to not mess with the display order since that could be set by the admins of the instalation. Maybe it would be better to bump up the offset to 1000.

Reply:
The ISO file itself seemed to be alphabetical. After one run of the code here, it looks like all of the current citation.tsv entries come first and then everything from the ISO file is in the order from that file. If there are changes in the ISO file though, the current code won't maintain it's internal order (either a duplicate displayOrder number as is, or the new entry will come after all existing entries if you update the code to query the db for number of entries (or max displayOrder) as in the todo.) Having the display order depend on when you run the api/if you ran it for intermediate changes in the ISO file, doesn't seem optimal. (Versus, for example, having a separate script to generate an always in sequence version of what would go in the citation.tsv file, or even just using only the line number from the ISO file as the displayOrder (or do we have citation.tsv entries that are not in the ISO file?)).